Quadriga/src/main/java/edu/asu/spring/quadriga/web/publicwebsite/NetworkSearchController.java

```diff
@@ -128,15 +128,39 @@ public String getSearch(@ProjectIdentifier @PathVariable("projectUnixName") Stri
         return new ResponseEntity<String>(defaultJsonErrorMsg, HttpStatus.INTERNAL_SERVER_ERROR);
     }
 
+    /**
+     * This method returns a display of the network that consists of all the statements in the
+     * given projects that contain the provided concept id.
+     * @param projectUnixName Unix name of the project whose networks will be searched for the concept
+     * @param conceptId Id of the concept
+     * @param project
+     * @param model
+     * @return relative path to the web-page that displays the network
+     * @throws QuadrigaStorageException
+     */
     @CheckPublicAccess
     @InjectProjectByName
     @RequestMapping(value = "sites/{projectUnixName}/networks/search", method = RequestMethod.GET)
     public String getSearchTransformedNetwork(@ProjectIdentifier @PathVariable("projectUnixName") String projectUnixName,
             @RequestParam("conceptId") String conceptId, @CheckAccess @InjectProject IProject project, Model model)
                     throws QuadrigaStorageException {
-
+        
+        String lemma = "";
+        String searchNodeLabel = "";
+        
+        // Fetch ConceptPower entries related to the conceptId
+        ConceptpowerReply reply = conceptpowerConnector.getById(conceptId);
+        List<String> alternativeIdsForConcept = null;
+        if (reply != null && reply.getConceptEntry().size() > 0) {
+            searchNodeLabel = reply.getConceptEntry().get(0).getLemma();
+            lemma = reply.getConceptEntry().get(0).getDescription();
+            alternativeIdsForConcept = reply.getConceptEntry().get(0).getAlternativeIdList();
+        }
```

Comment:
concepts are cached, so it's not a big deal if the conceptconnector is called twice.